src/providers/azure.rs

```diff
@@ -0,0 +1,503 @@
+use crate::ssh;
```

Comment:
I'd like mod-level documentation here that describes what this provider does, and its requirements (for example, it requires that the `az` command be in the user's `$PATH`).

Reply:
Sure, `aws` could probs use module-level docs as well